### PR TITLE
[MusicGen] Add sampling rate to config

### PR DESCRIPTION
**Pull request by sanchit-gandhi**
# What does this PR do?

Adds sampling rate attribute to the parent model config. We get the sampling rate from the EnCodec sub-model config. Adding this attribute brings consistency with the other TTA models in the lib, where the output sampling rate is accessible directly through the top-level config (https://github.com/huggingface/audio-transformers-course/pull/140#discussion_r1324226366)

### CODE REVIEW

File: tests/pipelines/test_pipelines_text_to_audio.py [hunk 1]
```diff
@@ -39,37 +39,59 @@ class TextToAudioPipelineTests(unittest.TestCase):
     model_mapping = MODEL_FOR_TEXT_TO_WAVEFORM_MAPPING
     # for now only test text_to_waveform and not text_to_spectrogram
 
+    @require_torch
+    def test_tiny_musicgen_pt(self):
+        music_generator = pipeline(
+            task="text-to-audio",
+            model="hf-internal-testing/tiny-random-MusicgenForConditionalGeneration",
+            framework="pt",
+        )
```
Comment (ylacombe) [inline tests/pipelines/test_pipelines_text_to_audio.py:48]: I might be wrong, but models that are in `hf-internal-testing` should be tested with `run_pipeline_test` called by [`PipelineTesterMixin`](https://github.com/huggingface/transformers/blob/32ec7345f2d752c294ddf5aff495b657c9cd9d3b/tests/test_pipeline_mixin.py#L122), right ?

Maybe @ydshieh can answer this question ?

**Review by ylacombe — APPROVED** [review]
I just have a small question regarding testing the tiny musicgen, but otherwise thanks for taking care of this @sanchit-gandhi and for the clearer renaming !

File: tests/pipelines/test_pipelines_text_to_audio.py (hunk 1 above)
Comment (sanchit-gandhi) [reply tests/pipelines/test_pipelines_text_to_audio.py:48]: Good spot - MusicGen currently isn't tested by the pipeline, but I'll fix this in a follow-up PR

File: tests/pipelines/test_pipelines_text_to_audio.py [hunk 2]
```diff
@@ -41,35 +41,35 @@ class TextToAudioPipelineTests(unittest.TestCase):
...
             ["This is a test", "This is a second test"], forward_params=forward_params, batch_size=2
         )
-        self.assertEqual(ANY(np.ndarray), outputs[0]["audio"])
+        audio = [output["audio"] for output in outputs]
+        self.assertEqual([ANY(np.ndarray), ANY(np.ndarray)], audio)
 
     @slow
     @require_torch
-    def test_large_model_pt(self):
+    def test_small_bark_pt(self):
```
Comment (ArthurZucker) [inline tests/pipelines/test_pipelines_text_to_audio.py:72]: 🙄

File: tests/pipelines/test_pipelines_text_to_audio.py [hunk 3]
```diff
@@ -41,35 +41,35 @@ class TextToAudioPipelineTests(unittest.TestCase):
...
-        # musicgen sampling_rate is not straightforward to get
-        self.assertIsNone(outputs["sampling_rate"])
-
-        audio = outputs["audio"]
-        self.assertEqual(ANY(np.ndarray), audio)
+        outputs = music_generator("This is a test", forward_params=forward_params)
+        self.assertEqual(
+            {"audio": ANY(np.ndarray), "sampling_rate": 32000},
+            outputs,
+        )
```
Comment (ArthurZucker) [inline tests/pipelines/test_pipelines_text_to_audio.py:56]:
```suggestion
        self.assertEqual({"audio": ANY(np.ndarray), "sampling_rate": 32000},outputs)
```

**Review by ArthurZucker — APPROVED** [review]
Looks good, good catch!

**Pull request merged**